x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/NodeDeprecationChecks.java

```diff
@@ -54,6 +54,22 @@ static DeprecationIssue auditLogPrefixSettingsCheck(List<NodeInfo> nodeInfos, Li
         return null;
     }
 
+    static DeprecationIssue auditIndexSettingsCheck(List<NodeInfo> nodeInfos, List<NodeStats> nodeStats) {
+        List<String> nodesFound = nodeInfos.stream()
+             .filter(nodeInfo -> (nodeInfo.getSettings().getByPrefix("xpack.security.audit.outputs").isEmpty() == false)
+                    || (nodeInfo.getSettings().getByPrefix("xpack.security.audit.index").isEmpty() == false))
```

Comment:
This seems to produce a deprecation issue if the `outputs` is non-empty, even if it only contains `logfile`.
Is that what you intended? It seems strange to deprecate _any_ explicit configuration of the output type.